RxSwift/Observables/Implementations/Merge.swift

```diff
@@ -56,12 +56,32 @@ fileprivate final class MergeLimitedSinkIter<S: ObservableConvertibleType, O: Ob
     }
 }
 
-fileprivate final class MergeLimitedSink<S: ObservableConvertibleType, O: ObserverType>
+fileprivate final class MapConcatSink<SourceType, S: ObservableConvertibleType, O: ObserverType>: MergeLimitedSink<SourceType, S, O> where O.E == S.E {
+    typealias Selector = (SourceType) throws -> S
+    
+    private let _selector: Selector
+    
+    init(selector: @escaping Selector, observer: O, cancel: Cancelable) {
+        _selector = selector
+        super.init(maxConcurrent: 1, observer: observer, cancel: cancel)
+    }
+    
+    override func performMap(_ element: SourceType) throws -> S {
+        return try _selector(element)
+    }
+}
+
+fileprivate final class MergeLimitedBasicSink<S: ObservableConvertibleType, O: ObserverType>: MergeLimitedSink<S, S, O> where O.E == S.E {
+    
+    override func performMap(_ element: S) throws -> S {
+        return element
+    }
+}
+
+fileprivate class MergeLimitedSink<SourceType, S: ObservableConvertibleType, O: ObserverType>
```

Comment:
Hi @sergdort ,

yeah, I think it's ok to make `MergeLimitedSink` a bit more generic and then reuse the logic.

I've reordered project structure a bit in `develop`. Can you please add unit tests for this and target `develop` branch?

We can reuse and modify `flatMap` unit tests a bit.